lxd/tokens.go

```diff
@@ -28,7 +31,44 @@ func autoRemoveExpiredTokens(ctx context.Context, s *state.State) {
 		}
 	}
 
-	if len(expiredTokenOps) == 0 {
+	isLeader, _, err := s.LeaderInfo()
+	if err != nil {
+		logger.Error("Failed to get database leader details", logger.Ctx{"err": err})
+	}
+
+	var expiredPendingTLSIdentities []cluster.Identity
+	if isLeader {
+		var pendingTLSIdentities []cluster.Identity
+		err := s.DB.Cluster.Transaction(ctx, func(ctx context.Context, tx *db.ClusterTx) error {
```

Comment:
Please can we move this part into a separate function called `getExpiredPendingIdentities` or similar?

Reply:
Done